celery/result.py

```diff
@@ -497,7 +497,6 @@ class ResultSet(ResultBase):
 
     def __init__(self, results, app=None, ready_barrier=None, **kwargs):
         self._app = app
-        self._cache = None
```

Comment:
my concern is removing cache attrib won;t cause any regression?

Reply:
Grepping the source indicates that the only place a `._cache` appears to be used is in this type's own `.get`, and in `backends/async.py`.  The latter appears irrelevant however, as in that case it is referring to `AsyncResult._cache` ... and the current `ResultSet._cache` is in an incompatible format to `AsyncResult._cache` so can't be being accessed there.

(`AsyncResult._cache` stores the complete metadata of the result, whereas `ResultSet._cache` stores just the return value ... which is the problem, you can't do that unless every single result in the list actually succeeded).

Further, users of `ResultSet` are just as likely to be accessing results via `join` as `get` ... and only `get` attempts to access `_cache` ... all the other methods for extracting results ignore it completely.

Reply:
redis integration tests are failing